pkgs/applications/misc/mainsail/default.nix

```diff
@@ -1,26 +1,33 @@
 { lib
-, stdenvNoCC
-, fetchzip
+, buildNpmPackage
+, fetchFromGitHub
 }:
 
-stdenvNoCC.mkDerivation rec {
+ buildNpmPackage rec {
   pname = "mainsail";
   version = "2.11.2";
 
-  src = fetchzip {
-    url = "https://github.com/mainsail-crew/mainsail/releases/download/v${version}/mainsail.zip";
-    hash = "sha256-RdBgGE/EUzb1/6PjQ34UjXjxt686s9May7npFtRocXE=";
-    stripRoot = false;
+  src = fetchFromGitHub {
+      owner = "mainsail-crew";
+      repo = "mainsail";
+      rev = "v${version}";
+      hash = "sha256-N0tm36YMRRrkyuIwzcYbDo1DHesAnJ2s2g0KCms3h5I=";
   };
 
-  dontConfigure = true;
-  dontBuild = true;
+  npmDepsHash = "sha256-z6Fo0XAds/F0Ig+nUE3O16gmH0EVcpML3K8cdKhkJzg=";
+
+  env.CYPRESS_INSTALL_BINARY = "0";
```

Comment:
Can you document what this does and why it's needed in a comment?

Reply:
I actually lifted this right out of [polaris](https://github.com/NixOS/nixpkgs/blob/cc6d856e654f54d3d40994bb0420ec9003a9ac9b/pkgs/servers/polaris/web.nix#L19), but as far as I can tell it just prevents the binary from attempting to download during the build phase.